bin/API_Migration_scripts_readme/WebScrapper_Java_Code-OfficialCopy.ipynb

```diff
@@ -0,0 +1,783 @@
+    "                        text = f.read()\n",
+    "                        if classes in text:\n",
+    "                            print(f\"Found {classes} in {java_file}\")    \n",
+    "    \n",
+    "                            \n",
+    "if __name__ == \"__main__\":\n",
+    "    match_csv_java(\"/Users/nseptank/Downloads/WebScrapper_Java_Code/links.csv\",\"/Users/nseptank/e-mission-phone/plugins/cordova-plugin-em-datacollection/src/android/location\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6fad8583",
+   "metadata": {},
+   "source": [
+    "#### location -> actions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "8deb408e",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Found java.util.Arrays in GeofenceLocationIntentService.java\n",
+      "Found java.util.Arrays in GeofenceActions.java\n",
+      "Found java.util.Collection in GeofenceLocationIntentService.java\n",
+      "Found android.content.Context in GeofenceActions.java\n",
+      "Found android.content.Context in LocationTrackingActions.java\n",
+      "Found android.content.Context in ActivityRecognitionActions.java\n",
+      "Found android.content.Intent in GeofenceLocationIntentService.java\n",
+      "Found android.content.Intent in GeofenceActions.java\n",
+      "Found android.content.Intent in LocationTrackingActions.java\n",
+      "Found android.content.Intent in ActivityRecognitionActions.java\n",
+      "Found android.content.IntentFilter in GeofenceActions.java\n",
+      "Found org.json.JSONObject in GeofenceActions.java\n",
+      "Found android.location.Location in GeofenceLocationIntentService.java\n",
+      "Found android.location.Location in GeofenceActions.java\n",
+      "Found java.util.concurrent.TimeUnit in GeofenceActions.java\n"
+     ]
+    }
+   ],
+   "source": [
+    "def match_csv_java(csv_file, directory):\n",
+    "    java_files = os.listdir(directory)\n",
+    "    with open(csv_file, \"r\") as f:\n",
+    "        reader = csv.reader(f)\n",
+    "        for row in reader:\n",
+    "            classes = row[1]         \n",
+    "    #java_files = os.listdir(directory)\n",
+    "            for java_file in java_files:\n",
+    "                if not os.path.isdir(java_file) and java_file.endswith(\".java\"):\n",
+    "                    with open(os.path.join(directory, java_file), \"r\", encoding=\"utf-8\") as f:\n",
+    "                        text = f.read()\n",
+    "                        if classes in text:\n",
+    "                            print(f\"Found {classes} in {java_file}\")    \n",
+    "    \n",
+    "                            \n",
+    "if __name__ == \"__main__\":\n",
+    "    match_csv_java(\"/Users/nseptank/Downloads/WebScrapper_Java_Code/links.csv\",\"/Users/nseptank/e-mission-phone/plugins/cordova-plugin-em-datacollection/src/android/location/actions\")"
+   ]
```

Comment:
Why do you have to call `match_csv_java` over and over again? Why isn't this just also an iteration over all the plugins?

Reply:
I go over each directory and iterate over the files. I tried iterating over the main directory but it doesn't seem to account for all the directories.

Reply:
What do you mean by "it doesn't account for all the directories"?

I tried this code:

```
...

```

and the output was 

```
....
```

Reply:
The script that matches the classes doesn't account for the number of directories inside the main directory (plugins). Furthermore, it would also account for other plugins. Rather than this approach, I set the directory to the one that contain the java files.

Reply:
I don't understand the comment "doesn't account for the number of directories inside the main directory (plugins)". Please see the template requested above. Concretely, if you walk the directory trees, they should find the directories at any level. 

> Furthermore, it would also account for other plugins. 

We do need to eventually make sure that all plugins work, so I see this as a plus

> Rather than this approach, I set the directory to the one that contain the java files.

Did you consider the disadvantages of this approach? I can think of at least two off the top of my head.